src/modules/ekf2/EKF/aid_sources/range_finder/range_finder_consistency_check.hpp

```diff
@@ -41,43 +41,68 @@
 #define EKF_RANGE_FINDER_CONSISTENCY_CHECK_HPP
 
 #include <mathlib/math/filter/AlphaFilter.hpp>
+#include <ekf_derivation/generated/state.h>
+
 
 class RangeFinderConsistencyCheck final
 {
 public:
 	RangeFinderConsistencyCheck() = default;
 	~RangeFinderConsistencyCheck() = default;
 
-	void update(float dist_bottom, float dist_bottom_var, float vz, float vz_var, bool horizontal_motion, uint64_t time_us);
+	enum class KinematicState : int {
+		INCONSISTENT = 0,
+		CONSISTENT = 1,
+		UNKNOWN = 2
+	};
 
-	void setGate(float gate) { _gate = gate; }
+	float getTestRatioLpf() const { return _initialized ? _test_ratio_lpf.getState() : 0.f; }
+	float getInnov() const { return _initialized ? _innov : 0.f; }
+	float getInnovVar() const { return _initialized ? _innov_var : 0.f; }
+	bool isKinematicallyConsistent() const { return _state == KinematicState::CONSISTENT; }
+	bool isNotKinematicallyInconsistent() const { return _state != KinematicState::INCONSISTENT; }
+	void setGate(const float gate) { _gate = gate; }
+	void run(const float z, const float vz, const matrix::SquareMatrix<float, estimator::State::size> P,
+		 const float dist_bottom, const float dist_bottom_var, uint64_t time_us);
+	void reset()
+	{
+		_state = (_initialized && _state == KinematicState::CONSISTENT) ? KinematicState::UNKNOWN : _state;
+		_initialized = false;
+	}
 
-	float getTestRatio() const { return _test_ratio; }
-	float getSignedTestRatioLpf() const { return _signed_test_ratio_lpf.getState(); }
-	float getInnov() const { return _innov; }
-	float getInnovVar() const { return _innov_var; }
-	bool isKinematicallyConsistent() const { return _is_kinematically_consistent; }
+	uint8_t current_posD_reset_count{0};
+	float terrain_process_noise{0.0f};
+	bool horizontal_motion{false};
 
 private:
-	void updateConsistency(float vz, uint64_t time_us);
 
-	uint64_t _time_last_update_us{};
+	void update(const float z, const float z_var, const float vz, const float vz_var, const float dist_bottom,
+		    const float dist_bottom_var, const uint64_t time_us);
+	void init(const float z, const float z_var, const float dist_bottom, const float dist_bottom_var);
+	matrix::SquareMatrix<float, 2> _R{};
+	matrix::SquareMatrix<float, 2> _P{};
+	matrix::SquareMatrix<float, 2> _A{};
+	matrix::SquareMatrix<float, 2> _H{};
+	matrix::Vector2f _x{};
+	bool _initialized{false};
+	float _innov{0.f};
+	float _innov_var{0.f};
+	uint64_t _time_last_update_us{0};
 	float _dist_bottom_prev{};
+	AlphaFilter<float> _test_ratio_lpf{0.2f};
+	float _gate{1.f};
+	int _sample_count{0};
+	KinematicState _state{KinematicState::UNKNOWN};
+	int _min_nr_of_samples{0};
```

Comment:
instead of computing how many samples are required and counting the number of samples you could simply accumulate the dt until it reaches the desired period (e.g.> 1s in your case).